ui/packages/components/src/Modal/Modal.tsx

```diff
@@ -30,7 +30,7 @@ export function Modal({
   footer,
   alignTop,
 }: ModalProps) {
-  const container = document?.getElementById('modals');
+  const container = typeof document !== 'undefined' ? document.getElementById('modals') : undefined;
```

Comment:
Also, unrelated: silence server side doc ref errors. This seems to be the canonical way to detect server side runtime execution as I see vercel themselves doing this all over their codebases. https://github.com/search?q=repo%3Avercel%2Fnext.js+%22typeof+window+%3D%3D%3D+%27undefined%27%22&type=code